Remove 3box feature and delete ThreeBoxController

rekmarks opened the pull request:
Removes the 3box sync feature and deletes the `ThreeBoxController`. Adds a migration to ensure that the `ThreeBoxController` is removed from persisted state.

This shaves 600kb or 3.2% from our bundle (18.5mb -> 17.9mb).

rekmarks committed:
Remove 3box feature and delete ThreeBoxController

Lint locale messages

lavamoat policy updates

kumavis commented:
:+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1:

darkwing reviewed (dismissed):
Oddly enough somehow the messages from `app/_locales/zh/messages.json` didn't get removed; I'm still seeing `syncWithThreeBox`, `syncWithThreeBoxDescription`, and `syncWithThreeBoxDisabled` keys.

brad-decker commented on app/scripts/controllers/metametrics.js:
```diff
@@ -569,7 +569,6 @@ export default class MetaMetricsController {
       ).length,
       [TRAITS.NUMBER_OF_TOKENS]: this._getNumberOfTokens(metamaskState),
       [TRAITS.OPENSEA_API_ENABLED]: metamaskState.openSeaEnabled,
-      [TRAITS.THREE_BOX_ENABLED]: metamaskState.threeBoxSyncingAllowed,
```
Could we, instead, set this to false and put a comment that it was set to false due to 3box removal? In a future PR we can remove it entirely but i'd like to see the user trait change gradually..If we delete this it'll be stuck in 'true' mode for those users who had it enabled before this is rolled out

Gudahtt replied:
Good idea! Done here: 46ab5a404450c4ec7b893afb5aa8fb6af42109cd

brad-decker commented on shared/constants/metametrics.js:
```diff
@@ -175,8 +175,6 @@
  * identify the new number_of_tokens trait
  * @property {'opensea_api_enabled'} OPENSEA_API_ENABLED - when the OpenSea API is enabled
  * we identify the opensea_api_enabled trait
- * @property {'three_box_enabled'} THREE_BOX_ENABLED - when 3box feature is
```
if my suggestion to turn the trait t ofalse instead of removing it is accepted, then keep these constants/types but notate they are deprecated.

Gudahtt replied:
Done: 3c3475bf20f1a2ac4445b30bc743632ca2969f23

brad-decker reviewed (dismissed):
Hero. 🙏🏻 Non blocking suggestion for metrics cleanliness.

Gudahtt commented:
We need a deprecation plan before we can merge this, including messaging to any existing users

darkwing commented:
Marking is `DO-NOT-MERGE` until we resolve Mark's comments.

kumavis commented:
@Gudahtt i would claim that we dont need a migration path bc the feature was only ever experimental. i agree that informing users that had it enabled is worth blocking on.

Gudahtt committed:
Restore 3Box user trait with value `false`

The 3Box user trait has been restored and hard-coded as `false`. This
ensures that users don't get stuck in our metrics as having this trait.

A deprecation comment has been left in various places for this trait.

Gudahtt committed:
Remove unused state

Gudahtt committed:
Remove additional 3box-related things

Gudahtt committed:
Run `yarn-deduplicate`

legobeat commented:
I assume https://github.com/MetaMask/metamask-extension/blob/794395dc48842385f9eecf5e3c2ae162b578d721/test/e2e/fixture-builder.js#L596 should go as well?  That'd be the last lines AFAICT.

Gudahtt committed:
Restore migration that was lost while rebasing

Gudahtt committed:
Remove obsolete override

Gudahtt committed:
Remove additional unused resolutions/dependencies

Gudahtt committed:
Update LavaMoat policies

Gudahtt committed:
Remove obsolete security advisory ignore entries

Gudahtt committed:
Remove 3Box fixture builder method

Gudahtt committed:
Update unit tests

danjm reviewed (approved):
LGTM
